### PR TITLE
feat: added faqs

### PR DESCRIPTION
**Pull request by AyushSehrawat**

### CODE REVIEW

**Comment by AyushSehrawat** [comment]
@mikBighne98 Review and let me know for any modifications. Also tell me the current limitations of space like allocated mem , size etc

**Comment by SlumberDemon** [comment]
**Deta Space is a serverless platform so it’s not something like a long running server that handles all the requests.**

I feel this is a bit of a mouthful, maybe rephrasing it might do the trick! For example:
> `Deta space is a serverless platform therefore it can't have long running servers for handling requests.`

**Instead it creates a new process for each request, responds to it and then that process gets released. ** 

Even though this is can be understood, I feel maybe it should be simplified just a little as not all people know what is meant by `released` or explain what is meant by it (that the resources used by the process are freed up and made available for reuse by other processes)

**Because Micros are serverless compute instances with a 20-second maximum lifespan per request, they do not support websockets.** 

I think `20 second` would be better than `20-second` because the previous information uses that format, it will help the consistency!

---

The rest is fine, with these minor adjustments this will be a good addition to the docs!

**Comment by AyushSehrawat** [comment]
> Even though this is can be understood, I feel maybe it should be simplified just a little as not all people know what is meant by released or explain what is meant by it (that the resources used by the process are freed up and made available for reuse by other processes)

Added "freed up" instead of released" The process stuff i handled by aws lambda and it's understandable to most. Can mention aws lambda working docs though

**Comment by AyushSehrawat** [comment]
@mikBighne98 Review it once and @SlumberDemon lmk if you have some faq

File: src/pages/docs/en/other/faq.md [hunk 1]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
 - [Discohook](https://github.com/jnsougata/discohook) (Discord)
 - [grammY](https://grammy.dev/) (Telegram)
 - [aiogram](https://github.com/aiogram/aiogram) (Telegram)
+
+## What are the limits of a Base?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:37]:
```suggestion
## What are the limits of Deta Base?
```

File: src/pages/docs/en/other/faq.md [hunk 2]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
...
+Bases currently do not have any limits but are meant for small to medium sized projects. If you are looking for a database for a large project, we recommend using some
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:38]:
```suggestion
Bases do not have a limit on the number of rows they can store, but each row is limited to 400 KB in size. Numbers are limited to 16 digits for both integers and floating-points.
```

File: src/pages/docs/en/other/faq.md [hunk 3]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
...
+other database or service. Also each document in a Base has a maximum size of 400kb.
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:39]:
```suggestion
```

File: src/pages/docs/en/other/faq.md [hunk 4]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
...
+
+## What are the limits of a Drive?
+Drives currently have a limit of 10GB per account. In the future we will be adding a way to increase this limit.
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:42]:
```suggestion
Drives currently have a global limit of 10 GB per account.
```

File: src/pages/docs/en/other/faq.md [hunk 5]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
 - [Discohook](https://github.com/jnsougata/discohook) (Discord)
 - [grammY](https://grammy.dev/) (Telegram)
 - [aiogram](https://github.com/aiogram/aiogram) (Telegram)
+
+## What are the limits of a Base?
+Bases currently do not have any limits but are meant for small to medium sized projects. If you are looking for a database for a large project, we recommend using some
+other database or service. Also each document in a Base has a maximum size of 400kb.
+
+## What are the limits of a Drive?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:41]:
```suggestion
## What are the limits of Deta Drive?
```

File: src/pages/docs/en/other/faq.md [hunk 6]
```diff
@@ -5,6 +5,26 @@ position: 0
...
+
+## Do Micros support websockets?
+Because Micros are serverless compute instances with a 20 seconds maximum lifespan per request, they do not support websockets.
+
+## My Micro timed out, what do I do?
+Your request took longer than 20 seconds to finish if your Micro timed out. You can try to speed up your code by optimising it or by putting another solution in place that doesn't need a constant connection.
+## Support for Python 3.11?
+Current micros include Python runtime support up to 3.10 and are powered by AWS Lambda. Micros will therefore support it once AWS Lambda begins to support those runtime.
+
+## What are the limits of a Micro?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:24]:
```suggestion
## What are the limits of Micros?
```

File: src/pages/docs/en/other/faq.md [hunk 7]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
...
 - [aiogram](https://github.com/aiogram/aiogram) (Telegram)
+
+## What are the limits of a Base?
+Bases currently do not have any limits but are meant for small to medium sized projects. If you are looking for a database for a large project, we recommend using some
+other database or service. Also each document in a Base has a maximum size of 400kb.
+
+## What are the limits of a Drive?
+Drives currently have a limit of 10GB per account. In the future we will be adding a way to increase this limit.
+
+## Domain verification pending, what do I do?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:44]:
```suggestion
## My domain verification is pending, what should I do?
```

File: src/pages/docs/en/other/faq.md [hunk 8]
```diff
@@ -13,3 +33,13 @@ Yes! Webhook-based bots that respond to HTTP requests can be easily deployed on
...
+It can take up to 24 hours for the domain to be verified. If it takes longer than that, make sure you followed the steps mentioned [here](docs/en/basics/micros#custom-domains) correctly. If you still face issues, please contact us at [Discord](https://go.deta.dev/discord).
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:45]:
```suggestion
Domains can take up to 24 hours to be verified. If it takes longer than that, make sure you have followed the steps mentioned [here](https://deta.space/docs/en/basics/micros#custom-domains) correctly. If you still face issues, please contact us on [Discord](https://go.deta.dev/discord).
```

File: src/pages/docs/en/other/faq.md [hunk 9]
```diff
@@ -5,6 +5,26 @@ position: 0
...
+
+## How do Micros work?
+Deta space is a serverless platform therefore it can't have long running servers for handling requests. Instead it creates a new process for each request, responds to it and then that process gets freed up. But the process has a timeout of 20 seconds. So if your request takes more than 20 seconds to complete, it will be terminated. 
+
+Lean more about Micros [here](/docs/en/basics/micros).
+
+## Do Micros support websockets?
+Because Micros are serverless compute instances with a 20 seconds maximum lifespan per request, they do not support websockets.
+
+## My Micro timed out, what do I do?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:19]:
```suggestion
## My Micro timed out, what should I do?
```

**Review by lemonyte — CHANGES_REQUESTED** [review]

File: src/pages/docs/en/other/faq.md (hunk 6 above)
Comment (AyushSehrawat) [reply src/pages/docs/en/other/faq.md:24]: Was to signify that each "Micro" has those limits rather than all micros ( like 5 micros in your app )

**Review by AyushSehrawat — COMMENTED** [review]
Added the required changes.

**Comment by AyushSehrawat** [comment]
@mikBighne98 Review once

**Comment by AyushSehrawat** [comment]
@mikBighne98 @lemonyte merge it when ?

**Review by SlumberDemon — APPROVED** [review]
Looks good to me!

File: src/pages/docs/en/other/faq.md [hunk 10]
```diff
@@ -5,6 +5,26 @@ position: 0
...
+## Why is Unauthorized error thrown when I try to access my app?
+By default all apps are private. You can make the app or parts of it public by editing the Spacefile and adding the `public_routes` key. Read more about it [here](/docs/en/basics/micros#public-routes).
+
+## How do Micros work?
+Deta space is a serverless platform therefore it can't have long running servers for handling requests. Instead it creates a new process for each request, responds to it and then that process gets freed up. But the process has a timeout of 20 seconds. So if your request takes more than 20 seconds to complete, it will be terminated. 
+
+Lean more about Micros [here](/docs/en/basics/micros).
+
+## Do Micros support websockets?
+Because Micros are serverless compute instances with a 20 seconds maximum lifespan per request, they do not support websockets.
```
[the remark this answers is not in the document]
Comment (lemonyte) [reply src/pages/docs/en/other/faq.md:14]: @AyushSehrawat please make this change, to follow standard English structure.

File: src/pages/docs/en/other/faq.md [hunk 11]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+By default all apps are private. You can make the entire app or parts of it public by adding the `public_routes` field to the Spacefile. Read more [in the docs](/docs/en/basics/micros#public-routes).
+
+## Do Micros support websockets?
+
+Because Micros are serverless compute instances with a 20 seconds maximum lifespan per request, they do not support websockets.
+
+## My Micro timed out, what should I do?
+
+A timeout error means your code took longer than 20 seconds to handle to a request. You can try to speed up your code by optimizing it or rework it so it doesn't need a constant connection
+## Support for Python 3.11?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:19]:
```suggestion

## Do Micros support Python 3.11?
```

File: src/pages/docs/en/other/faq.md [hunk 12]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+
+By default all apps are private. You can make the entire app or parts of it public by adding the `public_routes` field to the Spacefile. Read more [in the docs](/docs/en/basics/micros#public-routes).
+
+## Do Micros support websockets?
+
+Because Micros are serverless compute instances with a 20 seconds maximum lifespan per request, they do not support websockets.
+
+## My Micro timed out, what should I do?
+
+A timeout error means your code took longer than 20 seconds to handle to a request. You can try to speed up your code by optimizing it or rework it so it doesn't need a constant connection
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:18]:
```suggestion
A timeout error means your code took longer than 20 seconds to handle to a request. You can try to speed up your code by optimizing it or rework it so it doesn't need a constant connection.
```

File: src/pages/docs/en/other/faq.md [hunk 13]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+## Support for Python 3.11?
+
+Supported Python versions in Micros are determined by which versions are available on AWS Lambdas. When a new Python version is added to AWS Lambdas, it will be available shortly in Micros as well. You can check the current supported versions [here](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html).
+
+## Why am I encountering the HTTP error 413 (Entity Too Large) ?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:23]:
```suggestion
## Why am I encountering HTTP error 413 Entity Too Large?
```

File: src/pages/docs/en/other/faq.md [hunk 14]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+
+The HTTP error 413 (Entity Too Large) occurs when the request body surpasses our platform's maximum payload limit of 5 MB. To ensure efficient processing and prevent system overload, we recommend reviewing and reducing the size of your data. Splitting large data into smaller portions can help you stay within the payload limit and maintain optimal performance.
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:25]:
```suggestion
HTTP error 413 (Entity Too Large) occurs when the request body surpasses our platform's maximum payload limit of 5 MB. To ensure efficient processing and prevent system overload, we recommend reviewing and reducing the size of your data. Splitting large data into smaller chunks can help you stay within the payload limit and maintain optimal performance.
```

File: src/pages/docs/en/other/faq.md [hunk 15]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+
+## Why am I encountering the HTTP error 502 (Bad Gateway) ?
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:27]:
```suggestion
## Why am I encountering HTTP error 502 Bad Gateway?
```

File: src/pages/docs/en/other/faq.md [hunk 16]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+
+The HTTP error 502 (Bad Gateway) is typically encountered due to code issues or syntax errors that prevent the Micro from starting. Additionally, serving a file larger than 5 MB or adding a row to a Base that exceeds 400 KB can also cause this error. To resolve this issue, we recommend reviewing your code and ensuring that it is free of errors. If you are still encountering this error, please contact us on [Discord](https://go.deta.dev/discord).
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:29]:
```suggestion
HTTP error 502 (Bad Gateway) typically occurs due to code issues or syntax errors that prevent the Micro from starting. Additionally, serving a file larger than 5 MB or adding a row to a Base that exceeds 400 KB can also cause this error. To resolve this issue, we recommend reviewing your code and ensuring that it is free of errors. If you are still encountering this error, please contact us on [Discord](https://go.deta.dev/discord).
```

File: src/pages/docs/en/other/faq.md [hunk 17]
```diff
@@ -5,11 +5,51 @@ position: 0
...
+
+## What are the limits of a Micro?
+
+Our Space micro service is designed with specific resource limits to maintain efficient performance. Each request has a maximum lifespan of 20 seconds and is allocated a maximum of 512 MB of memory. Additionally, the payload size for a request is limited to 5 MB, while the package dependency size must not exceed 250 MB. It is important to note that exceeding any of these limits will result in the termination of the request.
```
Comment (lemonyte) [inline src/pages/docs/en/other/faq.md:33]:
```suggestion
Our Micro service is designed with specific resource limits to maintain efficient performance. Each request has a maximum lifespan of 20 seconds and is allocated a maximum of 512 MB of memory. Additionally, the payload size for a request is limited to 5 MB, while the package dependency size must not exceed 250 MB. It is important to note that exceeding any of these limits will result in the termination of the request.
```

**Review by lemonyte — CHANGES_REQUESTED** [review]

**Comment by abdelhai** [comment]
@AyushSehrawat would you like to accept the review suggestions?

**Comment by AyushSehrawat** [comment]
> @AyushSehrawat would you like to accept the review suggestions?

I will review them soon and then you can merge

**Comment by AyushSehrawat** [comment]
@lemonyte @abdelhai Done

**Review by lemonyte — APPROVED** [review]

**Comment by abdelhai** [comment]
@MaximilianHeidenreich could you please review & possibly merge

**Review by MaximilianHeidenreich — CHANGES_REQUESTED** [review]
Hey, thanks for this addition. I'll happily approve and merge it, but we need to migrate it to use Astro's content collections, as we do not have individual (content) pages inside the `pages` folder any longer. The current changes result in a build error: 
```
 error   `Astro.glob({})` did not return any matching files. Check the pattern for typos.
  File:
    /.../src/pages/docs/en/other/faq.md
```

For this to work, the `faq.md` file would need to be moved to an appropriate spot under `src/content/docs/`. Currently, I feel like this should actually not go inside the `Learn` section, as it really is targeted at developers content wise, BUT the concept of an FAQ page itself should go inside `Learn` (Maybe we can expand this with more user related information in the future) -> So I would suggest we place it there.

After moving it into the collection, the manifest file (`src/content/docs/_manifest.json`) needs a new `"faq.md"` entry at the corresponding location for it to show up inside the docs Sidebar.

Let me know if there is anything I can help with for this to be added.

**Comment by AyushSehrawat** [comment]
Moved to https://github.com/deta/space-docs/pull/180

**Pull request closed without merging**